sdks/unity/AgonesSdkClient.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json;
+using System.Collections;
+using System.Text;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace Agones.SDK
+{
+    /// <summary>
+    /// JSON payload object
+    /// </summary>
+    internal class KeyValueMessage

```

Comment:
Would using the existing `KeyValuePair<string, string>` class work for this? I believe it works with `JsonUtility`, but please double-check. Would save you defining a custom key value class though.